toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -71,124 +163,209 @@ func CustomizeImageWithConfigFile(buildDir string, configFile string, imageFile
 	return nil
 }
 
-func CustomizeImage(buildDir string, baseConfigPath string, config *imagecustomizerapi.Config, imageFile string,
-	rpmsSources []string, outputImageFile string, outputImageFormat string, outputSplitPartitionsFormat string,
-	useBaseImageRpmRepos bool, enableShrinkFilesystems bool,
-) error {
-	var err error
-	var qemuOutputImageFormat string
+func cleanUp(ic *ImageCustomizerParameters) error {
+
+	var allErrors error
 
-	outputImageBase := strings.TrimSuffix(filepath.Base(outputImageFile), filepath.Ext(outputImageFile))
-	outputImageDir := filepath.Dir(outputImageFile)
+	err := file.RemoveFileIfExists(ic.rawImageFile)
+	if err != nil {
+		allErrors = fmt.Errorf("failed to clean-up raw image file:\n%w", err)
+	}
 
-	// Validate 'outputImageFormat' value if specified.
-	if outputImageFormat != "" && outputImageFormat != ImageFormatIso {
-		qemuOutputImageFormat, err = toQemuImageFormat(outputImageFormat)
+	if ic.isoBuilder != nil {
+		err = ic.isoBuilder.cleanUp()
 		if err != nil {
-			return err
+			allErrors = fmt.Errorf("failed to clean-up iso builder:\n%w", err)
 		}
 	}
 
-	// Validate config.
-	err = validateConfig(baseConfigPath, config, rpmsSources, useBaseImageRpmRepos)
-	if err != nil {
-		return fmt.Errorf("invalid image config:\n%w", err)
-	}
+	return allErrors
+}
 
-	// Normalize 'buildDir' path.
-	buildDirAbs, err := filepath.Abs(buildDir)
+func CustomizeImage(buildDir string, baseConfigPath string, config *imagecustomizerapi.Config, imageFile string,
+	rpmsSources []string, outputImageFile string, outputImageFormat string, outputSplitPartitionsFormat string,
+	useBaseImageRpmRepos bool, enableShrinkFilesystems bool,
+) error {
+	err := validateConfig(baseConfigPath, config, rpmsSources, useBaseImageRpmRepos)
 	if err != nil {
-		return err
+		return fmt.Errorf("invalid image config:\n%w", err)
 	}
 
-	// Create 'buildDir' directory.
-	err = os.MkdirAll(buildDirAbs, os.ModePerm)
+	imageCustomizerParameters, err := createImageCustomizerParameters(buildDir, imageFile,
+		baseConfigPath, config,
+		useBaseImageRpmRepos, rpmsSources, enableShrinkFilesystems, outputSplitPartitionsFormat,
+		outputImageFormat, outputImageFile)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create image customizer:\n%w", err)
```

Comment:
nit - improve error message. append `parameters` to the existing message?

Reply:
pending